src/build/arg/arg_value.rs

```diff
@@ -0,0 +1,71 @@
+use std::fmt::Display;
+
+/// TODO
+#[derive(Debug, Default, Clone)]
+pub struct ArgValue<'help> {
+    pub(crate) value: &'help str,
+    pub(crate) about: Option<&'help str>,
+    pub(crate) hidden: bool,
+}
+
+impl Display for ArgValue<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.value.contains(char::is_whitespace) {
+            write!(f, "{:?}", self.value)
+        } else {
+            write!(f, "{}", self.value)
+        }
+    }
+}
+
+impl<'help> From<&'help str> for ArgValue<'help> {
+    fn from(s: &'help str) -> Self {
+        Self::new(s)
+    }
+}
+
+/// Getters
+impl<'help> ArgValue<'help> {
+    /// TODO
+    #[inline]
+    pub fn get_value(&self) -> &str {
+        self.value
+    }
+
+    /// TODO
+    #[inline]
+    pub fn get_about(&self) -> Option<&str> {
+        self.about
+    }
+
+    /// TODO
+    #[inline]
+    pub fn get_hidden(&self) -> bool {
+        self.hidden
+    }
+}
+
+impl<'help> ArgValue<'help> {
+    /// TODO
+    pub fn new<S: Into<&'help str>>(v: S) -> Self {
+        let value = v.into();
+        ArgValue {
+            value,
+            ..Default::default()
+        }
+    }
+
+    /// TODO
+    #[inline]
+    pub fn about<S: Into<&'help str>>(mut self, a: S) -> Self {
```

Comment:
Looks like clap generally doesn't use `Into<&str>`, so let's follow that there and just directly accept a `&str`.

In addition, when accepting any `str`, I believe the convention is `AsRef<str>` rather than `Into<&str>`

Reply:
Makes sense, I mirrored `Arg::new` here, but I can change it to either.